CMakeLists.txt

```diff
@@ -362,6 +362,20 @@ if(ENABLE_TESTING OR ENABLE_PROGRAMS)
         )
         add_custom_target(test_certs_header DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/tests/src/test_certs.h)
         add_dependencies(mbedtls_test test_keys_header test_certs_header)
+
```

Comment:
> I would put that rather in tests/CMakeLists.txt.

I wasn't sure, but the root file seemed to be a correct place since it's where we have the commands for `test_keys.h` and `test_certs.h`. Why is one location preferred? My understanding is that the normal thing in CMake is to have one `CMakeLists.txt` per directory that handles everything “in” that directory, but sometimes a build rule involves files outside that directory as well (e.g. files from the framework) and _sometimes_ that means the rule has to be in a parent directory of all the files involved. What goes into that last _sometimes_?

> tls13-compat.sh does not seem to be generated in the CMake case.

Indeed you have to run `make tls13-compat.sh` explicitly, which is a problem. What's the solution? `make generated_files` doesn't work.

Reply:
```
    set_target_properties(tls13-compat.sh PROPERTIES EXCLUDE_FROM_ALL NO)
```
seems to work. Please don't ask me why.